File: swagger/models/contact_log.py
from .base_model import BaseModel

from .client import Client
from .staff import Staff
from .staff import Staff
from .contact_log_comment import ContactLogComment
from .contact_log_type import ContactLogType

class ContactLog(BaseModel):
	"""
	A contact log.

	Attributes:

		id
		The contact log’s ID.

		text
		The contact log’s body text.

		created_date_time
		The local date and time when the contact log was created.

		followup_by_date
		The date by which the assigned staff member should close or follow up
		on this contact log.

		contact_method
		The method by which the client wants to be contacted.

		contact_name
		The name of the client to contact.

		client
		Information about the client to whom the contact log belongs.

		created_by
		Information about the staff member who created the contact log.

		assigned_to
		Information about the staff member to whom the contact log is assigned
		for follow up.

		comments
		The contact log’s comments.

		types
		Contains information about contact log types.
	"""

	input_map = {
		'Id': 'id',
		'Text': 'text',
		'CreatedDateTime': 'created_date_time',
		'FollowupByDate': 'followup_by_date',
		'ContactMethod': 'contact_method',
		'ContactName': 'contact_name',
		'Client': ('client', Client),
		'CreatedBy': ('created_by', Staff),
		'AssignedTo': ('assigned_to', Staff),
		'Comments': ['comments', ContactLogComment],
		'Types': ['types', ContactLogType],
		}

	output_map = {
		'id': 'Id',
		'text': 'Text',
		'created_date_time': 'CreatedDateTime',
		'followup_by_date': 'FollowupByDate',
		'contact_method': 'ContactMethod',
		'contact_name': 'ContactName',
		'client': 'Client',
		'created_by': 'CreatedBy',
		'assigned_to': 'AssignedTo',
		'comments': 'Comments',
		'types': 'Types',
		}

	id = None
	text = None
	created_date_time = None
	followup_by_date = None
	contact_method = None
	contact_name = None
	client = None
	created_by = None
	assigned_to = None
	comments = None
	types = None

File: swagger/Models/ContactLog.php
<?php

namespace App\Mindbody\Models;

/**
 * A contact log.
 */
class ContactLog extends BaseModel
{
	/**
	 * The contact log’s ID.
	 * 
	 * @var integer format:int64
	 */
	public $id;
	/**
	 * The contact log’s body text.
	 * 
	 * @var string 
	 */
	public $text;
	/**
	 * The local date and time when the contact log was created.
	 * 
	 * @var string format:date-time
	 */
	public $createdDateTime;
	/**
	 * The date by which the assigned staff member should close or follow up
	 * on this contact log.
	 * 
	 * @var string format:date-time
	 */
	public $followupByDate;
	/**
	 * The method by which the client wants to be contacted.
	 * 
	 * @var string 
	 */
	public $contactMethod;
	/**
	 * The name of the client to contact.
	 * 
	 * @var string 
	 */
	public $contactName;
	/**
	 * Information about the client to whom the contact log belongs.
	 * 
	 * @var Client 
	 */
	public $client;
	/**
	 * Information about the staff member who created the contact log.
	 * 
	 * @var Staff 
	 */
	public $createdBy;
	/**
	 * Information about the staff member to whom the contact log is assigned
	 * for follow up.
	 * 
	 * @var Staff 
	 */
	public $assignedTo;
	/**
	 * The contact log’s comments.
	 * 
	 * @var ContactLogComment[] 
	 */
	public $comments;
	/**
	 * Contains information about contact log types.
	 * 
	 * @var ContactLogType[] 
	 */
	public $types;

	protected function getInputMap()
	{
		return [
			'Id' => ['id', 'integer', 'int64'],
			'Text' => ['text', 'string', null],
			'CreatedDateTime' => ['createdDateTime', 'string', 'date-time'],
			'FollowupByDate' => ['followupByDate', 'string', 'date-time'],
			'ContactMethod' => ['contactMethod', 'string', null],
			'ContactName' => ['contactName', 'string', null],
			'Client' => ['client', Client::class, null],
			'CreatedBy' => ['createdBy', Staff::class, null],
			'AssignedTo' => ['assignedTo', Staff::class, null],
			'Comments' => ['comments', 'array', ContactLogComment::class],
			'Types' => ['types', 'array', ContactLogType::class],
			];
	}

	protected function getOutputMap()
	{
		return [
			'id' => ['Id', 'integer', 'int64'],
			'text' => ['Text', 'string', null],
			'createdDateTime' => ['CreatedDateTime', 'string', 'date-time'],
			'followupByDate' => ['FollowupByDate', 'string', 'date-time'],
			'contactMethod' => ['ContactMethod', 'string', null],
			'contactName' => ['ContactName', 'string', null],
			'client' => ['Client', Client::class, null],
			'createdBy' => ['CreatedBy', Staff::class, null],
			'assignedTo' => ['AssignedTo', Staff::class, null],
			'comments' => ['Comments', 'array', ContactLogComment::class],
			'types' => ['Types', 'array', ContactLogType::class],
			];
	}
}
